pages/contributing.md

```diff
@@ -1,216 +1,896 @@
-5. Developing unit tests
-   1. Use pytest to test, extend "unittest.TestCase" class for test cases
-   2. Unit tests should cover a variety of input object types and data types, for example GeoCAT-comp:
-      1. Test different combinations of numpy and xarray input
-      2. Test float64, float32, and any other relevant data types
-   3. For computational routines, we should ideally compare GeoCAT-comp output with known valid results from NCL
-   4. Small, synthetic, hard-coded datasets are ideal for unit tests so that they can run quickly, but some tests may require larger files, which should be accessed using GeoCAT-datafiles
-6. Python coding conventions
-   1. PEP8 with exceptions for line length
-      1. 99 chars for code (as described in PEP8 https://www.python.org/dev/peps/pep-0008/#maximum-line-length)
-      2. 72 characters for docstrings
-   2. We should add a linter to our repos as an extra check for PRs, same for code coverage
-   3. Function names?
-   4. No tabs, only spaces
-7. Documentation conventions?
-   1. Our repositories generally contain the source material for building documentation pages using sphinx, while some projects include the generated output as well
-8. Making use of available test data sets
-   1. Use files from geocat-datafiles when possible for either example scripts or unit tests that require datafiles
-   2. Avoid including any data files in source code repositories
-9. Repo-specific information
-   1. GeoCAT-comp
-      1. Computational functions and compiled code
-         1. Compiled code wrappers/cython will eventually be split out into a separate repo/package
-      2. Should I write Fortran or pure python?
-         1. Generally speaking, there is no reason a user needs to consider writing new functionality in Fortran; Fortran is only necessary when porting functions from NCL
-      3. Pure Python functionality should leverage the existence of other projects as appropriate
-         1. All functions should return xarray.DataArray objects
-         2. All functions should accept xarray.DataArray or numpy.ndarray as input, possibly also dask.da.Array
-         3. Allow for user-provided coordinate arrays instead of mandating usage of Xarray as input
-      4. New functionality should be treated as "experimental", which allows us to deploy it quickly and get it in the hands of users while making no guarantees about its interface or behavior
-      5. Can build documentation locally, but requires Cython code to be built (this will eventually be relevant for geocat-ncomp)
-   2. GeoCAT-datafiles
-      1. Need to run "python create_registry.py" to generate a new registry.txt whenever a new data file is added
-   3. GeoCAT-examples
-      1. Website can be built and tested locally
-   4. GeoCAT-viz
-      1. No documentation currently, we should consider adding some
+
+* Table of contents
+{:toc}
+
+Getting Started
```

Comment:
I'd number headings as we did in the latest version of this Contributor's Guide, because there are a lot of parts and headings, which make following and reading harder. Numbering would give the reader some hierarchy of those parts.

Reply:
The headings are numbered when the markdown is generated. In the top section, the tag number_sections: true takes care of it.